### ISSUE TITLE
Consider updating the version hash of luajit

### ISSUE THREAD

**Issue by adrinux** [ronoaldo/minetestserver#23]
For context see this issue I submitted to the [Mineclonia project](https://codeberg.org/mineclonia/mineclonia/issues/1627) which it appears may be due to the luajit version this Dockerfile uses and the fact I'm running on arm64.

The Dockerfile specifies a commit of luajit from February 2023 and comments that this is prior to the release of 2.1 -- luajit has now moved to a rolling release and 2.1 was marked 'production' in August 2023. There have been a number of arm64 related commits to luajit in that period and since.

The most recent commit to 2.1 is 04dca7911ea255f37be799c18d74c305b921c1a6

**Comment by ronoaldo** [comment]
Thank you for the update! I'm find upgrading to the more recent version of 2.1 branch, I would rather have a "released" version tag instead of a rolling one but it is what it is!

**Comment by adrinux** [comment]
I just experimented  with redefining the LUAJIT_VERSION in my Containerfile

```bash
# Use the default stable 5.x version
FROM ghcr.io/ronoaldo/minetestserver:stable-5

# Most recent LuaJIT commit on 2.1 branch as of 2024-06-06
ARG LUAJIT_VERSION=04dca7911ea255f37be799c18d74c305b921c1a6
```

But still end up with  a version `luajit-2.1.0-beta4-mercurio` in the container, so I assumed that hasn't worked.

It seems there is no way to override that without building my own base image too?
Wanted to do some testing.

**Comment by ronoaldo** [comment]
Yes... Unless you rebuild the luajit version in your base image, it won't rebuild what it inside the FROM clause, as each layer is immutable.

I'll try to add a branch and publish a version so you can confirm it works for you by pulling from a different tag. This way we can both test a newer version with our servers.

**Comment by adrinux** [comment]
I created a two stage build file based on yours and used the latest LuaJIT 2.1 (as in my last comment), rebuilt my containers and two of my worlds work fine (plain Minetest and Minetest plus a handful of mods in creative mod).

But Mineclonia was crashing on start.

I need to look a little deeper into why, and I can try a slightly older LuaJIT version, but I'm done for the day. Will let you know if I make any progress.

**Comment by adrinux** [comment]
Tried an older version of LuaJIT (ARG LUAJIT_VERSION=6807e60af13c7f9a582dbf4cabc56e57dc791808) and Mineclonia was no  longer crashing. But i was unable to mine blocks when I logged in and still found errors.

So instead of using Minetest master branch I specified a stable release (5.8.0). No crashes, seems to be working.

Right now I have this section in my Containerfile:

```docker
# Build-time arguments
ARG MINETEST_VERSION=5.8.0
ARG MINETEST_GAME_VERSION=5.8.0
# Note IRRLICHT is in main minetest repo from 5.9.0 onwards
ARG MINETEST_IRRLICHT_VERSION=1.9.0mt13
ARG MINETOOLS_VERSION=v0.2.2
# Most recent LuaJIT commit on 2.1 branch as of 2024-06-06
ARG LUAJIT_VERSION=04dca7911ea255f37be799c18d74c305b921c1a6
```

And this seems to work without errors - though  I have yet to stress test it with multiple players.

(Also note that irrlichtmt has been moved into the minetest repo from 5.9.0. https://github.com/minetest/irrlicht  is archived and pulling the head of master was probably  pulling in an old version as far as Minetest master is concerned -- it's possible that was ignored during the build -- but may also have been causing issues.)

**Comment by ronoaldo** [comment]
Because this images are not rebuilt frequently, I decided to go with branch v2.1 (rolling release of Lua) as a default when none of the build-time arguments are passed.

For the 5.9.0 release, I pinned a commit hash in order to keep this stable build consistent. This is done via the Github workflow. For downstream builds, if basing from 5.9.0 tags (including the updated stable-5 one), this hash is the one pinned.

There is no easy way to change linked libraries downstream unless rebuilding the container yourself with the proper --build-arg arguments.
